community/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/SpatialFunctionsAcceptanceTest.scala

```diff
@@ -185,6 +185,50 @@ class SpatialFunctionsAcceptanceTest extends ExecutionEngineFunSuite with NewPla
     result.toList should equal(List(Map("dist" -> null)))
   }
 
+  test("distance function should return null with null in any kind of point on rhs") {
+    var result = executeWithAllPlanners(
+      """return distance(point({latitude:3,longitude:7}),point({latitude:null,
```

Comment:
All tests are for distance(), but the implementation is actually on `point()`. It would be good to have a test on point function returning null. And then you also won't need as many tests.